### PR TITLE
Use post type support for enabling feature

### PR DESCRIPTION
**Pull request by grappler**

### CODE REVIEW

File: assets/js/src/editor/sidebar/edit.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { has, includes } from 'lodash';
+import { has, some } from 'lodash';
```
Comment (ocean90) [inline assets/js/src/editor/sidebar/edit.js]: `some` is unused

File: assets/js/src/editor/sidebar/edit.js [hunk 2]
```diff
@@ -119,16 +119,21 @@ function Edit( { media, headerImageId, onUpdateImage, onRemoveImage } ) {
...
-				select( 'core/editor' ).getEditedPostAttribute( 'type' )
-			),
+			postType: getPostType( getEditedPostAttribute( 'type' ) ),
 		};
 	} ),
-	ifCondition( ( { isPostTypePost } ) => isPostTypePost ),
+	ifCondition( ( { postType } ) => {
+		let isSupported = false;
+		if ( postType ) {
+			isSupported = postType.supports[ 'post-header-image' ];
```
Comment (ocean90) [inline assets/js/src/editor/sidebar/edit.js]: `isSupported` will be `undefined` or `null` if support is not set? Should make sure that it's always a boolean.

File: README.md [hunk 3]
```diff
@@ -10,9 +10,11 @@
 
 WordPress plugin for adding a custom header image to posts. Only supports the block editor. There is no support for the classic editor.
 
+To enable add post type support: `add_post_type_support( 'post', 'post-header-image' );`
```
Comment (ocean90) [inline README.md]: I think post should remain as default, similar to https://github.com/Automattic/jetpack/blob/792b26b5539d07cc35fdd93567942f2ad481eef2/modules/publicize/publicize.php#L787-L794

**Review by ocean90 — CHANGES_REQUESTED** [review]

File: assets/js/src/editor/sidebar/edit.js (hunk 2 above)
Comment (grappler) [reply assets/js/src/editor/sidebar/edit.js]: How would you do that?

Comment (derpaschi) [reply assets/js/src/editor/sidebar/edit.js]: Blauäugig wie ich bin, schlag ich mal sowas vor: 😂🙈
```suggestion
			isSupported = !! postType.supports[ 'post-header-image' ];
```

File: inc/namespace.php [hunk 4]
```diff
@@ -32,6 +33,13 @@ function register_post_meta() {
 	);
 }
 
+/**
+ * Ensures the Post internal post-type supports `post-header-image`.
```
Comment (ocean90) [inline inc/namespace.php]:
```suggestion
 * Ensures that the internal post type `post` supports `post-header-image`.
```

File: README.md [hunk 5]
```diff
@@ -10,9 +10,11 @@
 
 WordPress plugin for adding a custom header image to posts. Only supports the block editor. There is no support for the classic editor.
 
+To enable post-type support: `add_post_type_support( 'page', 'post-header-image' );`
```
Comment (ocean90) [inline README.md]:
```suggestion
By default, custom header images is only enabled for posts. To add support to other post types use the the 'add_post_type_support()` function. Example for pages:

	add_post_type_support( 'page', 'post-header-image' );
```

**Review by ocean90 — APPROVED** [review]

**Pull request merged**